docker/Dockerfile

```diff
@@ -0,0 +1,64 @@
+ARG BUILDER_IMAGE=python:3.9-slim-bullseye
+FROM ${BUILDER_IMAGE} as builder
+
+USER root
+WORKDIR /
+
+SHELL [ "/bin/bash", "-c" ]
+
+COPY . /root/madanalysis5
+
+# Set PATH to pickup virtualenv by default
+ENV PATH=/usr/local/venv/bin:"${PATH}"
+RUN apt-get -qq -y update && \
+    apt-get -qq -y install --no-install-recommends \
+      gcc \
+      g++ \
+      make \
+      zlib1g \
+      bash-completion \
+      python3-dev \
+      less \
+      tree \
+      wget \
+      curl \
+      gnuplot \
+      git && \
+    apt-get -y clean && \
+    apt-get -y autoremove && \
+    rm -rf /var/lib/apt/lists/* && \
+    python -m venv /usr/local/venv && \
+    . /usr/local/venv/bin/activate && \
+    python -m pip --no-cache-dir install --upgrade pip setuptools wheel && \
+    python -m pip --no-cache-dir install pip-tools && \
+    python -m pip list && \
+    python -m piptools compile \
+        --generate-hashes \
+        --output-file /root/madanalysis5/requirements.lock \
+        /root/madanalysis5/docker/requirements.txt && \
+    python -m pip --no-cache-dir install --upgrade --requirement /root/madanalysis5/requirements.lock && \
+    python -m pip list && \
+    export PATH="$(find / -type d -iname madanalysis5)/bin:${PATH}" && \
+    python -c 'import multiprocessing; print(multiprocessing.cpu_count())' | ma5 && \
```

Comment:
This works during the build, but when the image is run as a container later, `ma5` will warn that

```
MA5: Checking the MadAnalysis 5 core library:
MA5:   => System configuration has changed since the last use. Need to rebuild the library.
```

I'm not sure if there is a way to set defaults that will be general enough that a Docker image user won't have to have this compile step happen every time. Thoughts?

Reply:
I'll look into this. It's part of the legacy code; it has been built to be sensitive to any change.